Client/Application/LeanplumIntegration.swift

```diff
@@ -0,0 +1,196 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import AdSupport
+import Shared
+import Leanplum
+
+private let LeanplumEnvironmentKey = "LeanplumEnvironment"
+private let LeanplumAppIdKey = "LeanplumAppId"
+private let LeanplumKeyKey = "LeanplumKey"
+
+private enum LeanplumEnvironment: String {
+    case development = "development"
+    case production = "production"
+}
+
+enum LeanplumEventName: String {
+    case firstRun = "First Run"
+    case secondRun = "Second Run"
+    case openedApp = "Opened App"
+    case openedLogins = "Opened Login Manager"
+    case openedBookmark = "Opened Bookmark"
+    case openedNewTab = "Opened New Tab"
+    case interactWithURLBar = "Interact With Search URL Area"
+    case savedBookmark = "Saved Bookmark"
+    case openedTelephoneLink = "Opened Telephone Link"
+    case openedMailtoLink = "Opened Mailto Link"
+    case downloadedImage = "Download Media - Saved Image"
+    case closedPrivateTabsWhenLeavingPrivateBrowsing = "Closed Private Tabs When Leaving Private Browsing"
+    case closedPrivateTabs = "Closed Private Tabs"
+    case savedLoginAndPassword = "Saved Login and Password"
+    case userTappedFocusPromoButton = "User Tapped Focus Promo Button"
+    case focusPromoImpression = "Focus Promo Impression"
+    case focusPromoTimeout = "Focus Promo Timed Out"
+    case focusPromoTapDismiss = "Focus Promo was Dismissed By Tap"
+}
+
+private enum SupportedLocales: String {
+    case US = "en_US"
+    case DE = "de"
+    case UK = "en_GB"
+    case CA_EN = "en_CA"
+    case AU = "en_AU"
+    case TW = "zh_TW"
+    case HK = "en_HK"
+    case SG_EN = "en_SG"
+}
+
+private struct LeanplumSettings {
+    var environment: LeanplumEnvironment
+    var appId: String
+    var key: String
+}
+
+class LeanplumIntegration {
+    static let sharedInstance = LeanplumIntegration()
+
+    // Setup
+
+    private var profile: Profile?
+    private var enabled: Bool = false
+    
+    func shouldSendToLP() -> Bool {
+        return enabled && Leanplum.hasStarted()
```

Comment:
To get through this `privateMode` tracking problem, I'd suggest we have a `pause`/`unpause` metaphor, which is toggled when we're entering and leaving private browsing.

When paused, events are just dropped and never given to `Leanplum`.